### PR TITLE
Fixup Windows Dashboards and finish up adding Windows dashboards

### PR DESCRIPTION
**Pull request by aiyengar2**
Related Issue: https://github.com/rancher/rancher/issues/31921

### CODE REVIEW

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/workloads/rancher-workload.json [hunk 1]
```diff
@@ -0,0 +1,636 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/workloads/rancher-workload.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/workloads/rancher-workload-pods.json [hunk 2]
```diff
@@ -0,0 +1,636 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/workloads/rancher-workload-pods.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/windows/windows-metrics-summary.json [hunk 3]
```diff
@@ -1,768 +0,0 @@
-{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/windows/windows-metrics-summary.json:1]: now covered by rancher-cluster-nodes.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/windows/windows-metrics-detail.json [hunk 4]
```diff
@@ -1,768 +0,0 @@
-{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/windows/windows-metrics-detail.json:1]: now covered by rancher-cluster.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/pods/rancher-pod.json [hunk 5]
```diff
@@ -0,0 +1,620 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/pods/rancher-pod.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/pods/rancher-pod-containers.json [hunk 6]
```diff
@@ -0,0 +1,620 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/pods/rancher-pod-containers.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/nodes/rancher-node.json [hunk 7]
```diff
@@ -0,0 +1,776 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/nodes/rancher-node.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/nodes/rancher-node-detail.json [hunk 8]
```diff
@@ -0,0 +1,789 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/nodes/rancher-node-detail.json:1]: brand new dashboard

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/linux/linux-metrics-summary.json [hunk 9]
```diff
@@ -1,768 +0,0 @@
-{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/linux/linux-metrics-summary.json:1]: Now covered by rancher-cluster.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/linux/linux-metrics-detail.json [hunk 10]
```diff
@@ -1,768 +0,0 @@
-{
-    "annotations": {
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/linux/linux-metrics-detail.json:2]: now covered by rancher-cluster-nodes.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-k8s-components.json [hunk 11]
```diff
@@ -0,0 +1,502 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-k8s-components.json:1]: rename and refactor of kubernetes-components-metrics-summary.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-k8s-components-nodes.json [hunk 12]
```diff
@@ -0,0 +1,510 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-k8s-components-nodes.json:1]: rename and refactor of kubernetes-components-metrics-detail.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-etcd.json [hunk 13]
```diff
@@ -0,0 +1,652 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-etcd.json:1]: rename and refactor of etcd/etcd-metrics-summary.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-etcd-nodes.json [hunk 14]
```diff
@@ -0,0 +1,670 @@
+{
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/k8s/rancher-etcd-nodes.json:1]: rename and refactor of etcd/etcd-metrics-detail.json

File: packages/rancher-monitoring/generated-changes/overlay/files/rancher/home/rancher-default-home.json [hunk 15]
```diff
@@ -1,1275 +1,1273 @@
 {
-    "annotations": {
```
Comment (aiyengar2) [inline packages/rancher-monitoring/generated-changes/overlay/files/rancher/home/rancher-default-home.json:2]: Updated to contain everything that rancher/home/rancher-default-home-with-windows.json previously separately maintained and heavily refactored.

**Review by aiyengar2 — COMMENTED** [review]
On a high level, here are the changes introduced in this PR:
1. The directory structure of `files/rancher` (a folder that contains multiple subfolders, each of which consist of Rancher Original Grafana Dashboards) was modified and the corresponding ConfigMaps were modified to point to the new directory structure. 
2. Within the dashboard.jsons themselves, numerous changes were made throughout the files to fix up queries, titles, legendFormats, etc. and add templating variables for the relevant dashboards. I also modified all of the queries to query both windows + linux (except for 5 queries that did not have Windows counter-parts) since that was easier than maintaining a separate set of dashboards that have 95% similarity.

Since this PR is difficult to review via text, it would probably be better to view what the changes look like in a live cluster so I'm attaching screenshots of the dashboards deployed on a Windows cluster today.

### Home

Screen Shot 2021-04-19 at 3 21 47 PM

### Available Dashboards

Screen Shot 2021-04-19 at 3 23 48 PM

### Individual Dashboards
Screen Shot 2021-04-19 at 3 24 07 PM

Screen Shot 2021-04-19 at 3 24 56 PM

Screen Shot 2021-04-19 at 3 25 10 PM

Screen Shot 2021-04-19 at 3 25 23 PM

Screen Shot 2021-04-19 at 3 25 42 PM

Screen Shot 2021-04-19 at 3 25 55 PM

Screen Shot 2021-04-19 at 3 26 16 PM

Screen Shot 2021-04-19 at 3 26 44 PM

Screen Shot 2021-04-19 at 3 27 05 PM

Screen Shot 2021-04-19 at 3 27 25 PM

Screen Shot 2021-04-19 at 3 27 37 PM

Screen Shot 2021-04-19 at 3 27 54 PM

**Comment by aiyengar2** [comment]
Since the above screenshots don't capture everything that you might want to observe on Grafana (e.g. the legendFormats on hovering over each dashboards), lmk if I should spin up a live Windows cluster with this version of Monitoring installed for review.

**Comment by aiyengar2** [comment]
Note: Out of al of these dashboards, the only dashboard that has ever been previously released is the rancher-home-dashboard.json. Within the context of the 2.5.8 release, all other dashboards are brand new.

**Comment by dramich** [comment]
@aiyengar2 Can you summarize the differences between linux/windows dashboards? And how does that look in a cluster that has both? When you look at all nodes do some graphs just not have data for the windows nodes?

**Comment by aiyengar2** [comment]
> Can you summarize the differences between linux/windows dashboards? 

I refactored them so there are no differences between linux specific and windows specific dashboards. Instead, the PromQL queries using either the `{{__name__="node_*|windows_*"` mechanism to get the appropriate series from either windows or node or use a PromQL `OR` in order to either get the node_exporter query or, if empty, default to the windows-exporter query.

> And how does that look in a cluster that has both? 

In a cluster which has both windows and linux nodes, Windows hosts are treated as first class citizens. So it would look identical to any Linux host due to these changes.

> When you look at all nodes do some graphs just not have data for the windows nodes?

Only 1 graph has entirely no data for Linux nodes: Rancher / Pod + Rancher / Pod (Containers) => Disk I/O. I believe 2 metrics for Receive Error and Transmit Error are also missing under Rancher / Pod + Rancher / Pod (Containers) => Network Traffic.

But it will just show up as No Data for Windows hosts.

**Review by brendarearden — APPROVED** [review]
LGTM

**Review by dramich — APPROVED** [review]
